.github/workflows/eden.yml

```diff
@@ -57,26 +57,41 @@ jobs:
           TAG: pr${{ github.event.pull_request.number }}
           CACHE: evebuild/danger
         run: |
+          BUILD=true
           if docker pull "$CACHE:$TAG-kvm"; then
-             docker tag  "$CACHE:$TAG-kvm" "lfedge/eve:$TAG-kvm"
-             docker tag  "$CACHE:$TAG-kvm" "lfedge/eve:$TAG-kvm-amd64"
-          else
-             make -C eve V=1 PRUNE=1 pkgs
-             make -C eve V=1 ROOTFS_VERSION="$TAG" eve
-             IMAGES="$(docker images -f reference="lfedge/eve-*" -q)"
-             IMAGES="$IMAGES $(docker images -f reference="eve-build-*" -q)"
-             IMAGES="$IMAGES $(docker images -f reference="golang" -q)"
-             IMAGES="$IMAGES $(docker images -f dangling=true -q)"
-             docker rmi -f $IMAGES||echo "skip conflicts"
-             rm -rf ~/.linuxkit
+            # we should check version of pulled image
+            # since in case of build in progress we do not want to test
+            # old version pushed to docker hub
+            # version logic must be aligned with PR build workflow
+            COMMIT_ID=$(git --git-dir ./eve/.git describe --abbrev=8 --always)
+            EXPECTED_VERSION="0.0.0-$TAG-$COMMIT_ID-kvm-amd64"
+            PULLED_VERSION=$(docker run --rm "$CACHE:$TAG-kvm" version)
+            if [ "$PULLED_VERSION" = "$EXPECTED_VERSION" ]; then
+              docker tag "$CACHE:$TAG-kvm" "lfedge/eve:$TAG-kvm"
+              docker tag "$CACHE:$TAG-kvm" "lfedge/eve:$TAG-kvm-amd64"
+              BUILD=false
+            else
+              docker rmi --force "$CACHE:$TAG-kvm"
+            fi
+          fi
+          if [ "$BUILD" = "true" ]; then
+            make -C eve V=1 PRUNE=1 pkgs
+            make -C eve V=1 ROOTFS_VERSION="$TAG" eve
```

Comment:
Doesn't this make the workflow take a few hours longer? I wonder if it would be better to fail with a good error message so it can be manually re-run later.

Reply:
Well, It not expected to consume lot of time as we build only amd64 version on amd64 runner (and we still can run this with problems in arm64 build workflow we potentially want to solve separately). Also we have it now, I did not change this part.

Reply:
You're building all packages, which takes hours when I do that on my laptop. Do we currently have code in eden which rebuilds all of EVE-OS from source?

Reply:
Answering my own question, I see you just moved down the make code.
But isn't this going to run more often with these checks as opposed to only running when docker pull finds nothing?